guide/token-setup-and-validation.md

```diff
@@ -0,0 +1,351 @@
+            options.UseReferenceTokens();
+        })
+
+//Configure
+//must come before using MVC
+...
+app.UseAuthentication();
+...
+```
+
+## Reference token validation
+
+### Authorization server
+```c#
+// Startup.cs
+
+//ConfigureServices
+services.AddOpenIddict()
+        .AddCore(...) //see above
+
+        .AddServer(...) // see above
+
+        .AddValidation(options => options.UseReferenceTokens());
+
+//Configure
+//must come before using MVC
+...
+app.UseAuthentication();
+...
+```
+
+### Resource server
+```c#
+// Startup.cs
+
+//ConfigureServices
+// Register OpenIddict stores
+services.AddDbContext<AuthServerDbContext>(options =>
+    {
+        options.UseSqlServer(Configuration.GetConnectionString("AuthServerConnection"));
+        options.UseOpenIddict();
+    });
+
+
+services.AddOpenIddict()
+    .AddCore(options =>
+    {
+        // Register the Entity Framework entities and stores.
+        options.UseEntityFrameworkCore()
+               .UseDbContext<AuthServerDbContext>();
+    })
+
+    //This adds a "Bearer" authentication scheme
+    .AddValidation(options => options.UseReferenceTokens());
+
+//Optionally set Bearer token authentication as default
+//services.AddAuthentication("Bearer");
+
+//Configure
+//must come before using MVC
+...
+app.UseAuthentication();
+...
+```
+
+### Api controller
+```c#
+// Note: both OpenIddictValidationDefaults.AuthenticationScheme and JwtBearerDefaults.AuthenticationScheme are "Bearer"
+//If "Bearer" is not the default authentication scheme then specify it here.
+//If you get a 302 redirect to login page instead of a 401 Unauthorized then Cookie authentication is handling the request
+//so "Bearer" must be specified
+[Authorize(AuthenticationSchemes = "Bearer")]
+public class MyController : Controller
+```
+
+# JWTs
+
+## JWT generation
+
+### Authorization server
+```c#
+// Startup.cs
+
+//ConfigureServices
+services.AddOpenIddict()
+    .AddCore(...)
+    
+    .AddServer(options =>
+    {
+        //... configuration ommited for brevity
+        options.UseJsonWebTokens();
+        //JWTs must be signed by a self-signing certificate or a symmetric key
+        //Here a certificate is used. I used IIS to create a self-signed certificate
+        //and saved it in /FolderName folder. See below for .csproj configuration
+         options.AddSigningCertificate(
+                assembly: typeof(Startup).GetTypeInfo().Assembly,
+                resource: "AppName.FolderName.certname.pfx",
+                password: "anypassword");
+
+    })
+
+//Configure
+//must come before using MVC
+...
+app.UseAuthentication();
+...
+
+// .csproj
+// if using a certificate and its stored in your app's Resources then make sure it's published
+//<ItemGroup>
+//    <EmbeddedResource Include="FolderName\certname.pfx" />
+//  </ItemGroup>
+```
+
+## JWT validation
+
+### Authorization server
+```c#
+// Remember, this is only needed if you have api endpoints that require token authorization
```

Comment:
You should maybe move that to a specific `warn` block: https://dotnet.github.io/docfx/spec/docfx_flavored_markdown.html?tabs=tabid-1%2Ctabid-a#note-warningtipimportant